polynote-kernel/src/main/scala/polynote/kernel/ScalaCompiler.scala

```diff
@@ -521,43 +521,43 @@ object ScalaCompiler {
 
   def apply(
```

Comment:
So, this is getting a little bit ugly. Let me explain why there are now THREE different lists of JARs making up the classpath.

The first list is only the direct dependencies specified in the notebook config. The reason for this being separated out is primarily so that the class indexer (for completions) can prioritize matches based on whether it's a direct dependency (the idea being that you're more likely to want to use a class that you directly depended on than one you got transitively).

The second list is the transitive dependencies – any dependency that you didn't specify in the notebook config, but is needed because it's a dependency of one of your dependencies. Obviously we need these as well, and it's explained above why they're passed separately from the main dependencies.

The third list is all the dependencies that the *compiler needs to know about*, but which shouldn't be available to the dependency classloader. At this point, this is just Spark and its dependencies.